src/editor.c

```diff
@@ -1683,6 +1685,64 @@ static void close_block(GeanyEditor *editor, gint pos)
 /* checks whether @p c is an ASCII character (e.g. < 0x80) */
 #define IS_ASCII(c) (((unsigned char)(c)) < 0x80)
 
+/* Similar to read_current_word it will read the current scope of the word at the cursor position
+ * and writes it into the given buffer. The buffer will be NULL terminated in any case,
+ * even when the word is truncated because worlden is too small. psoition can be -1, then
+ * the current position is used.
+ * wc are the wordchars to use, if NULL, GEANY_WORDCHARS will be used */
+static void read_current_scope(GeanyEditor *editor, gint pos, gchar *word, gsize wordlen,
+		const gchar *wc)
+{
+	gint line, line_start, startword, endword;
+	gchar *chunk;
+	ScintillaObject *sci;
+
+	g_return_if_fail(editor != NULL);
+	sci = editor->sci;
+
+	if (pos == -1)
+		pos = sci_get_current_position(sci);
+
+	line = sci_get_line_from_position(sci, pos);
+	line_start = sci_get_position_from_line(sci, line);
+	startword = pos - line_start;
+	endword = pos - line_start;
+
+	word[0] = '\0';
+	chunk = sci_get_line(sci, line);
+
+	const char *punctuation = ":.";
```

Comment:
Well it doesn't assume LHS is the scope, it just checks for exact matches to display functions. If there are no matches it behaves normally.

But yeah it could lookup the type of any functions on the left hand side and convert it to an existing scope to pass to tm_workspace_find. Now that wouldn't be useful for me as I'm using it for javascript, and functions don't have static return types. But a quick test shows it it possible--
`class A {};
class B { static A aFunction(); };
B::aFunction`
Geany says: `aFunction's scope: B, class: (null), type: A`